### ISSUE TITLE
AudioParams in Panner and Listener

### ISSUE THREAD

**Issue by raub** [LabSound/LabSound#84]
According to the [spec](https://developer.mozilla.org/en-US/docs/Web/API/AudioListener/positionX) such properties as `AudioListener::positionX` must be of type `AudioParam` with gettable/settable `value`.

In LabSound only direct `void setPosition(float x, float y, float z)` is available. And this is actually deprecated, because of [reasons](https://developer.mozilla.org/en-US/docs/Web/API/AudioListener#Deprecated_features).

Is there any plan on converting those into `AudioParam`s?

**Comment by meshula** [comment]
Haven't looked at the panner node for a little while. That would be a reasonable update.

**Comment by raub** [comment]
In Chromium repo there's a lot of new stuff on this.

I guess it'll take a while to implement it in LabSound. For now I'm going to make some workaround to mimic settable AudioParams for JS side.

**Comment by meshula** [comment]
Complete: 

commit https://github.com/LabSound/LabSound/commit/471e59f6d157bc0851bb2f404bf1381b4779b3f8

The LabSound implementation is currently a superset of WebAudio, as they have simplified their spatialization scheme. This change is compatible, if you leave velocity, speed of sound, and doppler at default values.

**Comment by raub** [comment]
Oh, if only you saw that [workaround](https://github.com/raub/node-waa/blob/master/cpp/lab-audio-listener-param.cpp) 😆

Ok, now I'll pull the whole update and see if everything fits.

A question, though... How do you know param limits of +-10000 are sufficient? In my workaround I used std infinity, what are cons of that?

**Comment by meshula** [comment]
The webaudio implementation is a weird mix of double and float, probably one of the reasons they've changed it so much. infinity won't work well for that reason. I made a new issue. https://github.com/LabSound/LabSound/issues/87 

I set the limits to +/- one million until 87 is resolved. 

https://github.com/LabSound/LabSound/commit/ba6724802a89c39af14088f38aac3b891f6a5e30

**Comment by raub** [comment]
So, only AudioListener has the positional params currently? I don't see the params in [PannerNode](https://github.com/LabSound/LabSound/blob/master/include/LabSound/core/PannerNode.h).

**Comment by meshula** [comment]
That's right, however, the listener's params are used in the PannerNode implementation, that's the implementation work that needs fixing.

**Comment by raub** [comment]
If fixing the node is so distinct, lets move it into a separate issue (instead of reopenning this one).

The point is, until PannerNode doesn't have its params as well, I still have hacks for it in my code. And as long as I have those hacks, the issue persists.

**Comment by meshula** [comment]
Right you are. Missed that. https://github.com/LabSound/LabSound/commit/8a1fb3d521f8f54b23c13ce39d09df44148a28d9

**Comment by raub** [comment]
One more [thing](https://developer.mozilla.org/en-US/docs/Web/API/PannerNode/orientationX): now there is `forwardX`-etc, but it should be `orientationX`-etc.

**Comment by meshula** [comment]
Yeah. It's gross that listener and panner use a different word for the same thing. https://developer.mozilla.org/en-US/docs/Web/API/AudioListener
